### PR TITLE
server: reduce the overhead of calculating bytes flow

### PR DESCRIPTION
**Pull request by nolouch**
## What have you changed? (required)

Now PD will calculate the total bytes flow once region heartbeat coming, this is an extra cost. After this change, only calculate the bytes flow when store heartbeat coming. 
## What are the type of the changes (required)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (required)?

unit test

### CODE REVIEW

File: server/core/store.go [hunk 1]
```diff
@@ -327,7 +327,9 @@ func NewStoreNotFoundErr(storeID uint64) errcode.ErrorCode {
 
 // StoresInfo is a map of storeID to StoreInfo
```
Comment (rleungx) [inline server/core/store.go]: better to modify the comment

**Comment by siddontang** [comment]
can it affect the accuracy and immediacy of scheduler?

What overhead can it reduce?

**Comment by nolouch** [comment]
@siddontang This modification is mainly related to the heartbeat. We need to calculate hotspots according to heartbeat information, and pprof show it spends about 17% flat.

File: server/core/region_tree.go [hunk 2]
```diff
@@ -28,8 +28,8 @@ type regionItem struct {
 
 // Less returns true if the region start key is less than the other.
 func (r *regionItem) Less(other btree.Item) bool {
-	left := r.region.GetStartKey()
-	right := other.(*regionItem).region.GetStartKey()
+	left := r.region.StartKey
+	right := other.(*regionItem).region.StartKey
```
Comment (disksing) [inline server/core/region_tree.go]: Will this modification bring performance gains?

Comment (nolouch) [reply server/core/region_tree.go]: I will bench it and send another PR.

**Pull request merged**